azure-iot-device/azure/iot/device/common/auth/sastoken.py

```diff
@@ -94,3 +94,49 @@ def _build_token(self):
     def expiry_time(self):
         """Expiry Time is READ ONLY"""
         return self._expiry_time
+
+
+class NonRenewableSasToken(object):
+    def __init__(self, sastoken_string):
+        self._token = sastoken_string
+        self._token_info = get_sastoken_info_from_string(self._token)
+
+    def __str__(self):
+        return self._token
+
+    @property
+    def expiry_time(self):
+        """Expiry Time is READ ONLY"""
+        return int(self._token_info["se"])
+
+    @property
+    def resource_uri(self):
+        "Resource URI is READ ONLY"
+        uri = self._token_info["sr"]
+        return urllib.parse.unquote(uri)
+
+
+REQUIRED_SASTOKEN_FIELDS = ["sr", "sig", "se"]
+VALID_SASTOKEN_FIELDS = REQUIRED_SASTOKEN_FIELDS + ["skn"]
+
+
+def get_sastoken_info_from_string(sastoken_string):
+    pieces = sastoken_string.split("SharedAccessSignature ")
```

Comment:
>split [](start = 29, length = 5)

interesting use of the split function.  Are there cases where something can become before "SharedAccessSignature "?  I think I've only seen this as a prefix.

Reply:
Yeah, I don't believe so based on the SAS spec. I suppose I could have used .lstrip() here instead and it might have been clearer what I was trying to do.

---
In reply to: [506681987](https://github.com/Azure/azure-iot-sdk-python/pull/692#discussion_r506681987) [](ancestors = 506681987)